lib/friends/location.rb

```diff
@@ -10,11 +10,13 @@ class Location
     extend Serializable
 
     SERIALIZATION_PREFIX = "- "
+    # NOTE: nickname is the location alias, but `alias` is a keyword
+    NICKNAME_REFIX = "a.k.a. "
 
     # @return [Regexp] the regex for capturing groups in deserialization
     def self.deserialization_regex
       # Note: this regex must be on one line because whitespace is important
-      /(#{SERIALIZATION_PREFIX})?(?<name>.+)/
+      /(#{SERIALIZATION_PREFIX})?(?<name>[^\(\[@]*[^\(\[@\s])(\s+\(#{NICKNAME_REFIX}(?<nickname_str>.+)\))?/ # rubocop:disable Layout/LineLength
```

Comment:
I'm not sure I understand all of this—what is the `@` stuff for?

Reply:
That's right. The `@` stuff is not needed also `\[` is not needed. Because i first failed to create the right regex, i build it with the help of the deserialization_regex of friends.

So it is a bit shorter `[^\(]*[^\(\s]`.
Which then should be all character until ` (`. 

I tried a bit, but currently i didn't find a shorter solution. But i'm not so deep in regex may there is a way.

Reply:
So the way I think about this is we want the line in `friends.md` to match one of the following:

```
- New York City
- New York City (a.k.a. NYC)
- New York City (a.k.a. NYC a.k.a. the Big Apple a.k.a. Gotham)
```

to me that means it has the following:

- `- ` (the serialization prefix)
- zero or more characters that's not a `(`
- a last character in the name that's not a `(` or ` `
- (optionally) ` (` followed by `a.k.a. ` (the alias prefix) followed by one or more characters and then `)`

which I think would translate to:

```ruby
/(#{SERIALIZATION_PREFIX})?(?<name>[^\(]*[^\(\s])(\s+\(#{ALIAS_PREFIX}(?<alias_str>.+)\))?/
```

so in `initialize`, `name` would be `New York City` and `alias_str` would be `nil`, `NYC`, or `NYC a.k.a. the Big Apple a.k.a. Gotham`

Reply:
Yeah correct that works. My answer was a bit short. It referenced only the part i wanted to change, but not where it is.